### PR TITLE
Fix NEST and Elastisearch.NET v6 issue

### LINKED ISSUE #74: Elasticsearch 5.x
Hi,

Is there any plans to support Elasticsearch 5.x?

Thanks!

### PR DESCRIPTION
**Pull request by guibranco**
Fixed #74 issue

### CODE REVIEW

File: src/Elmah.Io.ElasticSearch/ElasticConnectionConfiguration.cs [hunk 1]
```diff
@@ -106,11 +106,7 @@ internal string ParseSingle(string connectionString, string key)
 
             var defaultIndexSegment = connectionStringSegments.FirstOrDefault(p => p.StartsWith(key, StringComparison.OrdinalIgnoreCase));
 
-            if (defaultIndexSegment == null)
-            {
-                return null;
-            }
-            return defaultIndexSegment.Substring(key.Length).TrimToNull();
+            return defaultIndexSegment?.Substring(key.Length).TrimToNull();
```
Comment (jayhilden) [inline src/Elmah.Io.ElasticSearch/ElasticConnectionConfiguration.cs]: Loving the new .net `?.` syntax!

File: src/Elmah.Io.ElasticSearch/ElasticConnectionConfiguration.cs [hunk 2]
```diff
@@ -172,10 +168,10 @@ internal static string GetDefaultIndex(IDictionary config, string connectionStri
         [Obsolete("this is here to support the ES connection string pre 1.2 release")]
         internal static string GetDefaultIndexFromConnectionString(string connectionString)
         {
-            Uri myUri = new Uri(connectionString);
+            var myUri = new Uri(connectionString);
```
Comment (jayhilden) [inline src/Elmah.Io.ElasticSearch/ElasticConnectionConfiguration.cs]: +1.  We can probably delete this obsolete method now that we're 2 releases beyond 1.2.  But that can be fore another day.

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec [hunk 3]
```diff
@@ -2,21 +2,23 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
   <metadata>
     <version>$version$</version>
-    <authors>elmah.io,jhilden</authors>
+    <authors>elmah.io,jhilden,Asgeir</authors>
```
Comment (jayhilden) [inline src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: nyoice!

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec [hunk 4]
```diff
@@ -2,21 +2,23 @@
...
       <dependency id="elmah" version="1.2.2" />
       <dependency id="elmah.corelibrary" version="1.2.2" />
-      <dependency id="NEST" version="2.4.5" />
+      <dependency id="NEST" version="6.2.0" />
+      <dependency id="Elasticsearch.Net" version="6.2.0" />
+      <dependency id="Newtonsoft.Json" version="11.0.2" />
     </dependencies>
-    <id>Elmah.ElasticSearch</id>
-    <title>Elmah.ElasticSearch</title>
+    <id>HRMTS.Elmah.ElasticSearch</id>
```
Comment (jayhilden) [inline src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: Please revert this change, we don't want the Nuget package to change.  Feel free to keep yourself as an author though.

**Review by jayhilden — CHANGES_REQUESTED** [review]
Super big thank you on this!  Just one change requested to the `.nuspec` file.  While you're doing that I'll hookup AppVeyor into the code review to make sure everything passes before the merge.

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec (hunk 4 above)
Comment (guibranco) [reply src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: Okay!

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec (hunk 3 above)
Comment (guibranco) [reply src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: I'll put my name too
```suggestion
    <authors>elmah.io,jhilden,Asgeir,gui.stracini</authors>
```

File: src/Elmah.Io.ElasticSearch/ElasticConnectionConfiguration.cs (hunk 2 above)
Comment (guibranco) [reply src/Elmah.Io.ElasticSearch/ElasticConnectionConfiguration.cs]: Removed!

**Comment by guibranco** [comment]
Changes done!

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec (hunk 3 above)
Comment (BernardTheDog) [inline src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: think my name can be removed ;)

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec (hunk 4 above)
Comment (BernardTheDog) [inline src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: project url to https://github.com/hrmts/Elmah.Io.ElasticSearch can also be removed...

File: src/Elmah.Io.ElasticSearch/ErrorDocument.cs [hunk 5]
```diff
@@ -1,42 +1,54 @@
...
 
         public string EnvironmentName { get; set; }
 
-        public string CustomerName { get; set; }
-    }
+        [Text(Index = true, Fielddata = true)]
+        public string CustomerName { get; set; }
+
+        [Text(Index = true, Fielddata = true)]
+        public string TenantId { get; set; }
```
Comment (BernardTheDog) [inline src/Elmah.Io.ElasticSearch/ErrorDocument.cs]: this tenantid i think i added, and can be removed too

File: src/examples/Elmah.Io.ElasticSearch.Web/Controllers/ElmahController.cs [hunk 6]
```diff
@@ -80,9 +80,8 @@ public override void ExecuteResult(ControllerContext context)
             var currentContext = GetCurrentContext(context);
 
             var httpHandler = factory.GetHandler(currentContext, null, null, null);
-            var httpAsyncHandler = httpHandler as IHttpAsyncHandler;
 
-            if (httpAsyncHandler != null)
```
Comment (BernardTheDog) [inline src/examples/Elmah.Io.ElasticSearch.Web/Controllers/ElmahController.cs]: sure this should be removed?

**Comment by jayhilden** [comment]
it looks like I've lost write access to this repo. @ThomasArdal can you please add me back in?

File: src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec (hunk 3 above)
Comment (guibranco) [reply src/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec]: You did the changes man, you don't want your name on the package ?

File: src/examples/Elmah.Io.ElasticSearch.Web/Controllers/ElmahController.cs (hunk 6 above)
Comment (guibranco) [reply src/examples/Elmah.Io.ElasticSearch.Web/Controllers/ElmahController.cs]: This is a enhancement due new C# version 7 feature called **pattern matching**
The old lines 83 (**cast**) and 85 (**if statement**) were merged with one expression in new 84 line using **pattern matching**!

`if(httpHandler is IHttpAsyncHandler httpAsyncHandler)` 

This statement do a cast and a null check in one time!

**Comment by jayhilden** [comment]
@guibranco AppVeyor should be hooked in now with the pull requests.  I cloned your branch and the build is failing.

https://ci.appveyor.com/project/jayhilden/elmah-io-elasticsearch/builds/19640001

Looks likely that a package is just not being pulled in via Nuget.

**Comment by guibranco** [comment]
Are you using a .yml file or set the AppVeyor to download de packages ? I didn't found any message of Nuget trying to restore the packages

Look another project with this message:
image

In this project I set the **appveyor.yml** to restore the packages before build

image

**Comment by jayhilden** [comment]
Yep, that's all in there

https://github.com/elmahio/Elmah.Io.ElasticSearch/blob/master/appveyor.yml#L13

**Comment by jayhilden** [comment]
you're 100% correct that it's not doing the nuget restore.  Here is the last successful build:

https://ci.appveyor.com/project/jayhilden/elmah-io-elasticsearch/builds/5485966

I'm going to have to play around and see if I can figure out why.  I haven't built into this project for 2 years so some things may have changed.

**Comment by jayhilden** [comment]
posted question to appveyor, waiting to hear back.

https://help.appveyor.com/discussions/problems/17583-appveyoryml-file-not-being-read

**Comment by jayhilden** [comment]
posted question to appveyor, waiting to hear back.

https://help.appveyor.com/discussions/problems/17583-appveyoryml-file-not-being-read

**Comment by ThomasArdal** [comment]
Just had a quick look in one of my appveyor.yml files. I can see that I had some problems getting nuget to restore as well in the past. Maybe try the following:

```
image: Visual Studio 2017
before_build:
- ps: dotnet restore
```

**Comment by ThomasArdal** [comment]
Just had a quick look in one of my appveyor.yml files. I can see that I had some problems getting nuget to restore as well in the past. Maybe try the following:

```
image: Visual Studio 2017
before_build:
- ps: dotnet restore
```

**Comment by ThomasArdal** [comment]
Just had a quick look in one of my appveyor.yml files. I can see that I had some problems getting nuget to restore as well in the past. Maybe try the following:

```
image: Visual Studio 2017
before_build:
- ps: dotnet restore
```

**Comment by ThomasArdal** [comment]
Just had a quick look in one of my appveyor.yml files. I can see that I had some problems getting nuget to restore as well in the past. Maybe try the following:

```
image: Visual Studio 2017
before_build:
- ps: dotnet restore
```

**Comment by ThomasArdal** [comment]
Just had a quick look in one of my appveyor.yml files. I can see that I had
some problems getting nuget to restore as well in the past. Maybe try the
following:

```
image: Visual Studio 2017
before_build:
- ps: dotnet restore
```

**Comment by guibranco** [comment]
Maybe if you add the line

```
image: Visual Studio 2017
```
to the appveyor.yml resolve the problem!

**Comment by guibranco** [comment]
Check the new commited file appveyor.yml that I've submited, I've added the image directive and fix the before_build instructions, adding "cmd:" before the nuget command like I do in other projects

**Comment by jayhilden** [comment]
Good news @guibranco!  I finally get appveyor back working, I cloned your PR branch and tweaked it a bit and now the build is working.

If you can please update this PR with the following appveyor.yml file then everything should be good to merge!

https://github.com/elmahio/Elmah.Io.ElasticSearch/blob/r3/appveyor.yml

[latest successful build on that branch](https://ci.appveyor.com/project/jayhilden/elmah-io-elasticsearch/builds/19720891)

**Comment by guibranco** [comment]
@jayhilden done!

**Comment by jayhilden** [comment]
:shipit: !!!

**Comment by guibranco** [comment]
There is a error with the building, looks like the Nuget key is invalid...

**Pull request merged**